utilities/pipelines/resourceRemoval/helper/Get-DeploymentTargetResourceList.ps1

```diff
@@ -108,7 +108,14 @@ function Get-DeploymentTargetResourceListInner {
                 [array]$resultSet = $resultSet | Where-Object { $_ -ne $deployment }
                 if ($deployment -match '/subscriptions/') {
                     # Subscription Level Child Deployments
-                    [array]$resultSet += Get-DeploymentTargetResourceListInner -Name (Split-Path $deployment -Leaf) -Scope 'subscription'
+                    if ($deployment -match '/resourceGroups/') {
```

Comment:
I'd like to understand this better. Wouldn't the passed in ID be first matching `/subscription/` and hence be passed into the recrusive function with scope `subscription` anyways where it then checks for `if ($deployment -match '/resourceGroups/')` the same way we do? 

I assume somewhere down the line the deploymentId is not properly resolved, but I wonder why.

Can you e.g. provide an example for a deployment ID which would not be properly passed down?

Reply:
Here is an example:

`/subscriptions/20d6fbfe-b049-471c-95af-1369d14d0d45/resourceGroups/validation-rg/providers/Microsoft.Resources/deployments/l4aytak3mfire-PolicyAssignment-RG-Module`

Flow:
1- scope = managementGroup
2- deploymentTarget = GetAzManagementGroupdeploymentOperation --> for deployment Name.. SUCCESS and returns the above ID
3- foreach deployment in that deploymentTarget
4- does it match /subscriptions/ --> success
5- Reruns the TargetResourceListInner at the subscriptionScope
6- deploymentTarget = Get-AzDeploymentOperation --> for deployment Name.. FAIL --> That does not exist at this level (only on Mgmt Group)

Hence it does not even reach the resource group level

Reply:
I'm not able to replicate the effect. I'd appreciate if we could have a chat next week to sort it out.